src/braket/circuits/compiler_directive.py

```diff
@@ -41,8 +43,43 @@ def ascii_symbols(self) -> Tuple[str, ...]:
         """Tuple[str, ...]: Returns the ascii symbols for the compiler directive."""
         return self._ascii_symbols
 
-    def to_ir(self, *args, **kwargs):
-        raise NotImplementedError("to_ir has not been implemented yet.")
+    def to_ir(
+        self,
+        target: QubitSet = None,
+        ir_type: IRType = IRType.JAQCD,
+        qubit_reference_format: str = "${}",
```

Comment:
Should we pass the Virtual/Physical enum in case there's ever any other differences (besides how the qubits are formatted) - maybe for future IRs?

Reply:
Alternatively we could pass an object that does the formatting of the qubit, rather than a string:

qubit_formatter = PhysicalQubitFormatter()

qubit_formatter.format(qubit)

?

Not sure if future-proof, or overly complex.

Reply:
We could add the additional argument when needed. The object seems a little heavy handed for now. We could extend the current parameter to accept `Union[str, QubitFormatter]` when we see the need.

Reply:
If it's a public method, and we allow string, should we validate if it's anything except for the two values that we support?

Reply:
Also, we should also make sure we document that the argument is unused if the ir_type is not OpenQASM... which is weird. It would be nicer if we could encapsulate the format with the type, since it's a feature of the format.